webapp/src/components/link_tooltip/link_tooltip.jsx

```diff
@@ -0,0 +1,191 @@
+import PropTypes from 'prop-types';
+import ReactMarkdown from 'react-markdown';
+import Octicon, {GitMerge, GitPullRequest, IssueClosed, IssueOpened} from '@primer/octicons-react';
+
+import Client from '../../client';
+import {validateGitlabURL} from '../../utils/regex_utils';
+import {isValidUrl} from '../../utils/url_utils';
+import './tooltip.css';
+
+const STATE_COLOR_MAP = {
+    OPENED_COLOR: '#28a745',
+    CLOSED_COLOR: '#cb2431',
+    MERGED_COLOR: '#6f42c1',
+};
+
+const STATE_TYPES = {
+    OPENED: 'opened',
+    CLOSED: 'closed',
+};
+
+const LINK_TYPES = {
+    MERGE_REQUESTS: 'merge_requests',
+    ISSUES: 'issues',
+};
+
+export const LinkTooltip = ({href, connected, gitlabURL}) => {
+    const [data, setData] = useState(null);
+    useEffect(() => {
+        if (!isValidUrl(href)) {
+            return;
+        }
+
+        const url = new URL(href);
+        const init = async () => {
+            if (url.origin === gitlabURL && validateGitlabURL(href)) {
+                const [owner, repo, , type, number] = href.split(`${url.hostname}/`)[1].split('/');
+                let res;
+                switch (type) {
+                case LINK_TYPES.ISSUES:
+                    res = await Client.getIssue(owner, repo, number);
+                    break;
+                case LINK_TYPES.MERGE_REQUESTS:
+                    res = await Client.getPullRequest(owner, repo, number);
+                    break;
+                }
+                if (res) {
+                    res = {...res, owner, repo, type};
+                }
+                setData(res);
+            }
+        };
+
+        if (data) {
+            return;
+        }
+
+        if (connected) {
+            init();
+        }
+    }, []);
+
+    const getIconElement = () => {
+        let color;
+        let iconType;
+        const {OPENED_COLOR, CLOSED_COLOR, MERGED_COLOR} = STATE_COLOR_MAP;
+        switch (data.type) {
+        case LINK_TYPES.MERGE_REQUESTS:
+            color = OPENED_COLOR;
+            iconType = GitPullRequest;
+            if (data.state === STATE_TYPES.CLOSED) {
+                if (data.merged) {
+                    color = MERGED_COLOR;
+                    iconType = GitMerge;
+                } else {
+                    color = CLOSED_COLOR;
+                }
+            }
+            break;
+        case LINK_TYPES.ISSUES:
+            color = data.state === STATE_TYPES.OPENED ? OPENED_COLOR : CLOSED_COLOR;
+            iconType = data.state === STATE_TYPES.OPENED ? IssueOpened : IssueClosed;
+            break;
+        }
+        const icon = (
+            <span style={{color}}>
+                <Octicon
+                    icon={iconType}
+                    size='small'
+                    verticalAlign='middle'
+                />
+            </span>
+        );
+        return icon;
+    };
+
+    if (data) {
+        const date = new Date(data.created_at).toDateString();
```

Comment:
To remove some indentation

```suggestion
    if (!data) {
        return null;
```